src/Nethermind/Nethermind.Logging.NLog/NLogManager.cs

```diff
@@ -12,140 +12,147 @@
 using NLog.Targets;
 using Level = NLog.LogLevel;
 
-namespace Nethermind.Logging.NLog
+namespace Nethermind.Logging.NLog;
+
+public class NLogManager : ILogManager
 {
-    public class NLogManager : ILogManager
+    private const string DefaultFileTargetName = "file-async_wrapped";
+    private const string DefaultFolder = "logs";
+
+    public NLogManager(string logFileName, string logDirectory = null, string logRules = null)
     {
-        private const string DefaultFileTargetName = "file-async_wrapped";
+        Setup(logFileName, logDirectory, logRules);
+        // Required since 'NLog.config' could change during runtime, we need to re-apply the configuration
+        LogManager.ConfigurationChanged += (sender, args) => Setup(logFileName, logDirectory, logRules);
```

Comment:
*What are the best practices when dealing with this kind of event handler?*

I've read that you could use `LogManager.ConfigurationChanged = null` to remove all handlers, but what if we attach a new one on other parts of the code?

Ideally, we would only remove this handler during `Shutdown`.